plutus-contract/src/Wallet/Emulator/Chain.hs

```diff
@@ -68,14 +68,20 @@ makeLenses ''ChainState
 
 data ChainControlEffect r where
     ProcessBlock :: ChainControlEffect Block
+    ModifySlot :: (Slot -> Slot) -> ChainControlEffect Slot
```

Comment:
Why did you choose this to take a function rather than a final value? Do we have any concurrency issues?

Reply:
I wanted to avoid having to call `getCurrentSlot` every time `ModifySlot` is called... but maybe just `PutSlot Slot` would be better.